membership-attribute-service/test/filters/AddGuIdentityHeadersTest.scala

```diff
@@ -27,14 +28,17 @@ class AddGuIdentityHeadersTest extends Specification with IdiomaticMockito {
     authTime = None,
   )
 
-  val identityService = mock[IdentityAuthService]
   val request = mock[RequestHeader]
-  when(identityService.user(requiredScopes = Nil)(request)).thenReturn(Future.successful(Right(user)))
   val config = ConfigFactory.load()
   val testUsernames = CreateTestUsernames.from(config)
-  val isTestUser = new IsTestUser(testUsernames)
+  val testUserChecker = new TestUserChecker(testUsernames)
 
-  val addGuIdentityHeaders = new AddGuIdentityHeaders(identityService, isTestUser)
+  def setup = {
+    val identityService = mock[IdentityAuthService]
+    when(identityService.user(requiredScopes = Nil)(request)).thenReturn(Future.successful(Right(user)))
+    val addGuIdentityHeaders = new AddGuIdentityHeaders(identityService, testUserChecker)
+    (identityService, addGuIdentityHeaders)
+  }
```

Comment:
interestingly this test was failing randomly because specs2 only creates one test instance per execution and runs the tests in parallel.  This means the mock identity service was sometimes called by the second test before it was asserted by the first.
This technique was suggested here: https://stackoverflow.com/a/56251734/2076522